test/test_plugins.py

```diff
@@ -531,3 +537,62 @@ def foo(self, session, task):
         self.mock_input_options.assert_called_once_with(
             opts, default="a", require=ANY
         )
+
+
+def get_available_plugins():
+    """Get all available plugins in the beetsplug namespace."""
+    namespace_pkg = importlib.import_module("beetsplug")
+    plugin_names: list[str] = []
+    for _, module_name, _ in pkgutil.iter_modules(namespace_pkg.__path__):
+        plugin_names.append(module_name)
+
+    # skip _typing
+    return [name for name in plugin_names if name != "_typing"]
+
+
+class TestImportAllPlugins:
+    @pytest.mark.parametrize("plugin_name", get_available_plugins())
+    def test_import_plugin(self, caplog, plugin_name):  #
+        """Test that a plugin is importable without an error using the
+        load_plugins function."""
+
+        # Try to load all plugins
+        from beets.plugins import load_plugins
+
+        caplog.set_level(logging.WARNING)
+        load_plugins([plugin_name])
+
+        # Check for warnings, is a bit hacky but we can make full use of the beets
+        # load_plugins code that way
+        records = []
+        pattern = r"ModuleNotFoundError: No module named '(.*?)'"
+        for record in caplog.records:
+            match = re.search(pattern, str(record))
+            if match:
+                module_name = match.group(1)
+                if not self._is_spec_available(module_name):
+                    # If the module is not found, we skip it
+                    continue
+            records.append(record)
+
+        assert len(records) == 0, (
+            f"Plugin '{plugin_name}' has issues during import. ",
+            records,
+        )
+
+        self.unimport_plugins()
+
+    def _is_spec_available(self, spec_name):
+        """Check if a module is available by its name."""
+        return importlib.util.find_spec(spec_name) is not None
+
+    def unimport_plugins(self):
```

Comment:
The test failures might be caused by not unloading plugins properly, as it seems like the tests only start failing after running this one.

To unload plugins, you don't just need to remove them from `sys.modules`, you also need to make beets forget about them.

Your best bet is to make this test class extend `beets.test.helper.PluginTestCase` and use `self.load_plugin(plugin_name)` and `self.unload_plugins()`.

Reply:
Ah thanks! I just reviewed #5872 which will hopefully fix this too. So in theory no change should be necessary here :crossed_fingers: